src/Files.App/Controllers/TerminalController.cs

```diff
@@ -15,198 +15,192 @@
+
+		public async Task InitializeAsync()
+		{
+			await LoadAsync();
+			await GetInstalledTerminalsAsync(model);
+			SaveModel();
+			await StartWatchConfigChangeAsync();
+			ModelChanged?.Invoke(this);
+		}
+
+		public void SaveModel()
+		{
+			try
+			{
+				var path = Path.Combine(ApplicationData.Current.LocalFolder.Path, "settings", JsonFileName);
+				using var file = File.CreateText(path);
+
+				// update local configContent to avoid unnecessary refreshes
+				configContent = JsonSerializer.Serialize(model, DefaultJsonSettingsSerializer.Options);
+				file.Write(configContent);
+			}
+			catch
+			{
+			}
+		}
+
+		private async Task LoadAsync()
+		{
+			StorageFolder Folder = await FilesystemTasks.Wrap(() =>
+				ApplicationData.Current.LocalFolder.CreateFolderAsync("settings", CreationCollisionOption.OpenIfExists)
+			.AsTask());
+
+			if (Folder is null)
+			{
+				model = await GetDefaultTerminalFileModel();
+				return;
+			}
+
+			var jsonFile = await FilesystemTasks.Wrap(() =>
+				Folder.GetFileAsync(JsonFileName)
+			.AsTask());
+
+			if (!jsonFile)
+			{
+				model = await GetDefaultTerminalFileModel();
+				if (jsonFile.ErrorCode is FileSystemStatusCode.NotFound)
+					SaveModel();
+				return;
+			}
+
+			try
+			{
+				configContent = await FileIO.ReadTextAsync(jsonFile.Result);
+
+				var fileModel = JsonSerializer.Deserialize<TerminalFileModel>(configContent);
+				if (fileModel is null)
+					throw new ArgumentException($"{JsonFileName} is empty, regenerating...");
+				model = fileModel!;
+			}
+			catch (ArgumentException)
+			{
+				model = await GetDefaultTerminalFileModel();
+				SaveModel();
+			}
+			catch (Exception)
+			{
+				model = await GetDefaultTerminalFileModel();
+			}
+		}
+
+		private async Task StartWatchConfigChangeAsync()
+		{
+			var queryOptions = new QueryOptions
+			{
+				ApplicationSearchFilter = "System.FileName:" + JsonFileName
+			};
+
+			var settingsFolder = await ApplicationData.Current.LocalFolder.GetFolderAsync("settings");
+			var query = settingsFolder.CreateFileQueryWithOptions(queryOptions);
+			query.ContentsChanged += Query_ContentsChanged;
+			await query.GetFilesAsync();
+		}
+
+		private async void Query_ContentsChanged(IStorageQueryResultBase sender, object args)
+		{
+			try
+			{
+				var uri = new Uri("ms-appdata:///local/settings/terminal.json");
+				var file = await StorageFile.GetFileFromApplicationUriAsync(uri);
+				var content = await FileIO.ReadTextAsync(file);
+
+				if (configContent == content)
+					return;
+
+				configContent = content;
+				await LoadAsync();
+				await GetInstalledTerminalsAsync(model);
+				SaveModel();
+				ModelChanged?.Invoke(this);
+			}
+			catch {}
+		}
+
+		private async Task<TerminalFileModel> GetDefaultTerminalFileModel()
+		{
+			try
+			{
+				var uri = new Uri(defaultTerminalPath);
+				var defaultFile = await StorageFile.GetFileFromApplicationUriAsync(uri);
+				var defaultContent = await FileIO.ReadTextAsync(defaultFile);
```

Comment:
Those lines are repetition of lines l.117 to 1.119 with different names for the variables. Worth creating a method of them both.

Reply:
I added a method.